Add last activity column to Student Management

Imran92 opened the pull request:
Part of https://github.com/Automattic/sensei/issues/4958

### Changes proposed in this Pull Request

Last activity date column is added to the student management table with sorting ability.

Moved the last activity date formating function to utility class because it was being used in multiple places.

### Testing instructions
1. Create a few courses
2. Create a few students
3. Enroll the students to those courses
4. Complete some lessons on different times/days
5. Go the SenseiLMS->Students
6. See if the last activity column is there
7. Try sorting it using the last activity column header
8. See if the similar behavior holds even when filtered by courses

### Screenshot / Video
![Screen Shot 2022-04-06 at 10 48 49 PM](https://user-images.githubusercontent.com/6820724/162026395-62b7d819-2888-4e3a-bd67-8649871fa8e6.png)

Commit:
Merge branch 'feature/student-management' into add/last-activity-to-student-management

Commit:
Move last activity date formatting function to Util class

Commit:
Add last activity date to student management with sorting

Commit:
Add test for last activity date in student management table

merkushin commented on includes/class-sensei-utils.php:
```diff
@@ -2581,7 +2581,32 @@ public static function output_query_params_as_inputs( array $excluded = [] ) {
+	 *
+	 * @return string The formatted last activity date.
+	 */
+	public static function format_last_activity_date( string $date ) {
```
What do you think about adding a test for this method?

merkushin reviewed (dismissed):
Works well 👍

Commit:
Merge branch 'feature/student-management' into add/last-activity-to-student-management

donnapep commented on includes/class-sensei-db-query-learners.php:
```diff
@@ -55,17 +55,23 @@ private function build_query() {
+					SELECT MAX(cm.comment_date_gmt)
+					FROM {$wpdb->comments} cm
+					WHERE cm.user_id = u.ID
+					AND cm.comment_approved IN ('complete', 'passed', 'graded')
```
~Should this also include "ungraded"? Does the Students report include that status?~ This date should match what's on the Students report, is that right?

UPDATE: As per my comment [here](https://github.com/Automattic/sensei/pull/4863#discussion_r814944834), looks like it shouldn't include it.

Imran92 replied:
Great! Thanks a lot,  I had a little confusion regarding the completion state, now it's cleared :)

donnapep replied:
But it should match the Students report, yes? 🙂

Commit:
Added tests for last activity formatting util function

Imran92 commented:
Thanks a lot @merkushin for the suggestions, lost the approved state due to a merge and adding some tests, can you kindly take a look again? I've added tests for the possible outcomes and fixed an issue in the Util testing class that was preventing it from using dataprovider for tests

merkushin reviewed (approved):
Thanks for adding tests!
